<a href="https://colab.research.google.com/github/Divarahman/pemprograman-komputer-5018211034/blob/Week02/W02_Engineering_Calculator_5018211034.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Week 2 Assignment
## Engineering Unit and Electrical Calculator
Nama: Diva Muhamad Rahman |
NRP: 5018211034 |
Kelas : Pemprograman Komputer

## Task 1: Speed Unit Conversion

In [9]:
speed_kmh = float(input("Enter speed [km/h]: "))
speed_ms = speed_kmh / 3.6
print(f"Speed = {speed_ms:.2f} m/s")

Enter speed [km/h]: 100
Speed = 27.78 m/s


## Task 2: Power Unit Conversion

In [10]:
power_kw = float(input("Power [kW]: "))
power_w = power_kw * 1000
print(f"Power = {power_w:.2f} W")

Power [kW]: 12.5
Power = 12500.00 W


## Task 3: Electrical Power and Energy Calculator

## Test Cases
Test 1: V=12, I=5.0, T=2.0

Test 2: V=24, I=8.5, T=3.0

Test 3: V=48, I=10.0, T=1.5







In [11]:
voltage = float(input("Voltage [V]: "))
current = float(input("Current [A]: "))
time = float(input("Operating Time [h]: "))

power = voltage * current
energy = power * time

print("===== ELECTRICAL CALCULATION =====")
print(f"Voltage        = {voltage:.2f} V")
print(f"Current        = {current:.2f} A")
print(f"Power          = {power:.2f} W")
print(f"Operating Time = {time:.2f} hour")
print(f"Energy         = {energy:.2f} Wh")

Voltage [V]: 48
Current [A]: 10
Operating Time [h]: 1.5
===== ELECTRICAL CALCULATION =====
Voltage        = 48.00 V
Current        = 10.00 A
Power          = 480.00 W
Operating Time = 1.50 hour
Energy         = 720.00 Wh


## Task 4: Comparison and Logical Operators

In [12]:
power_limit = 200
voltage_limit = 24

is_high_power = power > power_limit
is_high_voltage = voltage > voltage_limit

print(f"High power        = {is_high_power}")
print(f"High voltage      = {is_high_voltage}")
print(f"Both conditions   = {is_high_power and is_high_voltage}")
print(f"Any condition     = {is_high_power or is_high_voltage}")

High power        = True
High voltage      = True
Both conditions   = True
Any condition     = True


## 9. Short Reflection
1. Apa perbedaan int dan float?
   > Jawaban:
   int (integer) adalah tipe data untuk bilangan bulat tanpa komponen desimal (misalnya: 12, -5), sedangkan float (floating point) adalah tipe data untuk bilangan riil yang memiliki titik/pecahan desimal (misalnya: 12.5, 3.6).

2. Mengapa hasil input() perlu dikonversi menggunakan float() untuk perhitungan numerik?
   > Jawaban:
   Fungsi bawaan input() di Python selalu mengembalikan data dalam format teks atau string (str). Jika data masukan tersebut tidak dikonversi ke tipe data numerik seperti float(), operasi aritmatika (seperti perkalian atau pembagian) akan memicu TypeError, atau operasi penjumlahan + hanya akan menggabungkan teks (string concatenation) alih-alih menjumlahkan nilainya secara matematis.

3. Apa perbedaan operator = dan == dalam Python?
   > Jawaban:
   Operator = adalah assignment operator yang berfungsi untuk menetapkan atau menyimpan sebuah nilai ke dalam suatu variabel (contoh: power_limit = 200). Sedangkan operator == adalah comparison operator (operator perbandingan) yang menguji apakah dua nilai bernilai sama, dan menghasilkan nilai Boolean True atau False.

4. Apa fungsi operator and dan or?
   > Jawaban:
   Operator and bernilai True hanya jika kedua kondisi yang diuji bernilai True (jika salah satu False, hasilnya False). Sedangkan operator or bernilai True jika salah satu atau kedua kondisi yang diuji bernilai True (hanya menghasilkan False jika seluruh kondisi bernilai False).

5. Apa satu kesalahan yang Anda temui saat mengerjakan tugas ini, dan bagaimana Anda memperbaikinya?
   > Jawaban:
  Kesalahan terjadi saat lupa mengubah string dari input() menjadi angka numerik sebelum melakukan pembagian kecepatan di Task 1, yang menyebabkan program berhenti akibat TypeError. Kesalahan tersebut diperbaiki dengan menambahkan fungsi input() di dalam fungsi konversi float(), yaitu float(input(..)).